server/routerlicious/packages/services/src/tenant.ts

```diff
@@ -53,10 +53,18 @@ export class TenantManager implements core.ITenantManager {
 
     public async getTenant(tenantId: string, documentId: string, includeDisabledTenant = false): Promise<core.ITenant> {
         const restWrapper = new BasicRestWrapper();
-        const [details, key] = await Promise.all([
+        const [details, gitManager] = await Promise.all([
             restWrapper.get<core.ITenantConfig>(`${this.endpoint}/api/tenants/${tenantId}`,
             { includeDisabledTenant }),
-            this.getKey(tenantId, includeDisabledTenant)]);
+            this.getTenantGitManager(tenantId, documentId, includeDisabledTenant)]);
+
+        const tenant = new Tenant(details, gitManager);
+
+        return tenant;
+    }
+
+    public async getTenantGitManager(tenantId: string, documentId: string, includeDisabledTenant = false): Promise<GitManager> {
+        const key = await this.getKey(tenantId, includeDisabledTenant);
 
         const defaultQueryString = {
```

Comment:
Not directly related to this PR but I am curious as to why we need to send this token as a part of the querystring since we are already sending the token used for auth as a part of the header

Reply:
This is actually not related to auth. It seems to be just an additional way of sending the tenantId to Riddler. It has been added as part of [#5302](https://github.com/microsoft/FluidFramework/pull/5302) by Zach and is being used for [logging](https://github.com/microsoft/FluidFramework/blob/e4780ece326c3400175ea69a578f4e3479ae9dcf/server/routerlicious/packages/routerlicious-base/src/riddler/app.ts#L40) on Riddler side. In fact, the [logic](https://github.com/microsoft/FluidFramework/blob/e4780ece326c3400175ea69a578f4e3479ae9dcf/server/routerlicious/packages/routerlicious-base/src/utils/params.ts#L13) looks for tenantId in query params first and falls back to request path params otherwise.